src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,213 @@
+  SendableChooser<String> m_chooser = new SendableChooser<>();
+
+  private final Solenoid gearShift;
+
+// Creates a new Drivetrain.
+public Drivetrain() {
+
+    // Sets variables equal to their ports and types
+    m_leftMotor1 = new CANSparkMax(DriveConstants.kLeftDriveMotor1Port, MotorType.kBrushless);
+    m_leftMotor2 = new CANSparkMax(DriveConstants.kLeftDriveMotor2Port, MotorType.kBrushless);
+    m_rightMotor1 = new CANSparkMax(DriveConstants.kRightDriveMotor1Port, MotorType.kBrushless);
+    m_rightMotor2 = new CANSparkMax(DriveConstants.kRightDriveMotor2Port, MotorType.kBrushless);
+
+    motorInit(m_rightMotor1, DriveConstants.kRightEncoderInverted); 
+    motorInit(m_rightMotor2, DriveConstants.kRightEncoderInverted);
+    motorInit(m_leftMotor1, DriveConstants.kLeftEncoderInverted);
+    motorInit(m_leftMotor2, DriveConstants.kLeftEncoderInverted); 
+    m_leftMotors = new SpeedControllerGroup(m_leftMotor1, m_leftMotor2);
+    m_rightMotors = new SpeedControllerGroup(m_rightMotor1, m_rightMotor2);
+
+    m_Drive = new DifferentialDrive(m_leftMotors, m_rightMotors);
+
+     // adds options to the drive type
+     m_chooser.addOption("Tank Drive", "Tank");
+     m_chooser.addOption("Arcade Drive", "Arcade");
+     m_chooser.addOption("Trigger Drive", "Trigger");
+
+    m_leftMotorEncoder1 = m_leftMotor1.getEncoder();
+    m_leftMotorEncoder2 = m_leftMotor2.getEncoder();
+    m_rightMotorEncoder1 = m_rightMotor1.getEncoder();
+    m_rightMotorEncoder2 = m_rightMotor2.getEncoder();
+
+    gearShift = new Solenoid(DriveConstants.kDriveSolenoid);
+
+    addChild("Left Motors", m_leftMotors);
+    addChild("Right Motors", m_rightMotors);
+    addChild("Drivetrain", m_Drive);
+    addChild("Shift Gears", gearShift);
+
+  }
+
+private void encoderInit(CANEncoder encoder) {
+    // Converts the input into desired value for distance and velocity
+    encoder.setPositionConversionFactor(DriveConstants.kEncoderDistancePerPulse);
+    encoder.setVelocityConversionFactor(DriveConstants.kEncoderSpeedPerPulse);
+    encoderReset(encoder);
+  }
+
+private void encoderReset(CANEncoder encoder) {
+    encoder.setPosition(0);
+  }
+
+public void resetLeftEncoders() {
+    // Resets all left-side drive encoders
+    encoderReset(m_leftMotorEncoder1);
+    encoderReset(m_leftMotorEncoder2);
+  }
+
+public void resetRightEncoders() {
+    // Resets all right-side drive encoders
+    encoderReset(m_rightMotorEncoder1);
+    encoderReset(m_rightMotorEncoder2);
+  }
+
+public void resetAllEncoders() {
+    // Resets all drive encoders
+    resetLeftEncoders();
+    resetRightEncoders();
+  }
+
+private void motorInit(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults(); // Resets settings in motor in case they are changed
+    motor.setIdleMode(IdleMode.kBrake); // Sets the motors to brake mode from the beginning
+    motor.setSmartCurrentLimit(kCurrentLimit);
+
```

Comment:
This method should also be setting the motor to be inverted or not according to the parameter passed through to "invert"
```suggestion
motor.setInverted(invert);
```

Reply:
added the invert parameter to the method